verilog/analysis/verilog_project.cc

```diff
@@ -341,4 +341,50 @@ absl::StatusOr<FileList> ParseSourceFileListFromFile(
   return ParseSourceFileList(file_list_file, content);
 }
 
+absl::StatusOr<FileList> ParseSourceFileListFromCommandline(
+    const std::vector<char*>& cmdline) {
+  FileList file_list_out;
+  for (std::string argument :
+       verible::make_range(cmdline.begin() + 1, cmdline.end())) {
+    // cmdline[0] is the tool's name, which can be skipped.
+    if (argument[0] != '+') {
+      // Then "argument" is a SV file name.
+      file_list_out.file_paths.push_back(argument);
+    } else {
+      // it should be either a define or incdir.
+      std::vector<std::string> argument_plus_splitted =
+          absl::StrSplit(argument, absl::ByChar('+'), absl::SkipEmpty());
+      if (argument_plus_splitted.size() < 2) {
+        return absl::InvalidArgumentError("Unkown argument.");
+      }
+      std::string plus_argument_type = argument_plus_splitted[0];
+      if (absl::StrContains(plus_argument_type, "define")) {
```

Comment:
why 'contains' and not equal ? Otherwise we'd match `+foodefinebar+` which is not really what we want